src/coreclr/vm/interpexec.cpp

```diff
@@ -11,14 +11,36 @@
 // for numeric_limits
 #include <limits>
 
-#ifdef TARGET_WASM
-void InvokeCalliStub(PCODE ftn, CallStubHeader *stubHeaderTemplate, int8_t *pArgs, int8_t *pRet);
-void InvokeCompiledMethod(MethodDesc *pMD, int8_t *pArgs, int8_t *pRet, PCODE target);
+// Call invoker helpers provided by platform.
+void InvokeManagedMethod(MethodDesc *pMD, int8_t *pArgs, int8_t *pRet, PCODE target);
+void InvokeUnmanagedMethod(MethodDesc *targetMethod, int8_t *stack, InterpMethodContextFrame *pFrame, int32_t callArgsOffset, int32_t returnOffset, PCODE callTarget);
+void InvokeCalliStub(PCODE ftn, void* stubHeaderTemplate, int8_t *pArgs, int8_t *pRet);
 void InvokeDelegateInvokeMethod(MethodDesc *pMDDelegateInvoke, int8_t *pArgs, int8_t *pRet, PCODE target);
-#else
+
+// Use the NOINLINE to ensure that the InlinedCallFrame in this method is a lower stack address than any InterpMethodContextFrame values.
+NOINLINE
+void InvokeUnmanagedMethodWithTransition(MethodDesc *targetMethod, int8_t *stack, InterpMethodContextFrame *pFrame, int32_t callArgsOffset, int32_t returnOffset, PCODE callTarget)
+{
+    InlinedCallFrame inlinedCallFrame;
+    inlinedCallFrame.m_pCallerReturnAddress = (TADDR)pFrame->ip;
+    inlinedCallFrame.m_pCallSiteSP = pFrame;
+    inlinedCallFrame.m_pCalleeSavedFP = (TADDR)stack;
+    inlinedCallFrame.m_pThread = GetThread();
+    inlinedCallFrame.m_Datum = NULL;
+    inlinedCallFrame.Push();
+
+    {
+        GCX_PREEMP();
+        InvokeManagedMethod(targetMethod, stack + callArgsOffset, stack + returnOffset, callTarget);
```

Comment:
Why is this function called InvokeUnmanagedMethodWithTransition if it's calling InvokeManagedMethod?

Reply:
Because the current `InvokeManagedMethod`, previously `InvokeCompiledMethod` does double duty in the interpreter path. @jkotas made this suggestion offline.

Reply:
The current code makes assumption that the managed calling convention is same as the unmanaged calling convention. That's not a valid assumption. 

I expect that this will need to be redone to handle unmanaged calling conventions properly that you have touched upon on the teams chat. Once it is redone InvokeUnmanagedMwethod won't be calling InvokeManagedMethod anymore.

It may be a good idea to add `WASM-TODO: Handle unmanaged calling conventions` next to this to avoid confusion in the meantime.